app/test_main.py

```diff
@@ -1 +1,25 @@
-# write your code here
+from app.main import can_access_google_page
+import pytest
+from unittest import mock
+
+
+@pytest.mark.parametrize(
+    "google_url, internet_connection,result",
+    [
+        (True, True, "Accessible"),
+        (True, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (False, False, "Not accessible")
+    ]
+)
+def test_can_access_google_page(
+        result: str,
+        google_url: bool,
+        internet_connection: bool
+) -> None:
+    with mock.patch("app.main.valid_google_url", return_value=google_url):
+        with mock.patch(
+                "app.main.has_internet_connection",
+                return_value=internet_connection
+        ):
```

Comment:
Make sure your code is not very nested . Use more beautiful variant for example decoarator or something else which is less nested then now

Reply:
sry :(
 i did a new solution